composer/models/huggingface.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from torch import nn
+from torchmetrics import Metric
+from torchmetrics.collections import MetricCollection
+
+from composer.models.base import ComposerModel
+
+if TYPE_CHECKING:
+    import transformers
+
+__all__ = ["HuggingFaceModel"]
+
+
+class HuggingFaceModel(ComposerModel):
+    """
+    Args:
+        model ([`PreTrainedModel`] or `torch.nn.Module`):  You can  use your own models defined as
```

Comment:
nit: use `|` for type unions in docstrings

```suggestion
        model (transformers.PreTrainedModel | torch.nn.Module):  You can use your own models defined as
```